com.unity.render-pipelines.high-definition/Editor/Lighting/LightUnit/LightUnitSlider.cs

```diff
@@ -202,19 +240,141 @@ protected override float GetPositionOnSlider(float value)
             return ValueToSlider(value);
         }
 
-        protected override void DoSlider(Rect rect, SerializedProperty value, Vector2 range)
+        protected override void DoSlider(Rect rect, SerializedProperty value, Vector2 sliderRange)
         {
             value.floatValue = ExponentialSlider(rect, value.floatValue);
         }
 
         float ExponentialSlider(Rect rect, float value)
         {
-            var internalValue = GUI.HorizontalSlider(rect, ValueToSlider(value), 0f, 1f, GUI.skin.horizontalSlider, GUI.skin.horizontalSliderThumb);
-
+            var internalValue = GUI.HorizontalSlider(rect, ValueToSlider(value), 0f, 1f);
             return SliderToValue(internalValue);
         }
     }
 
+    /// <summary>
+    /// Formats the provided descriptor into a piece-wise linear slider with contextual slider markers, tooltips, and icons.
+    /// </summary>
+    class PiecewiseLightUnitSlider : LightUnitSlider
+    {
+        struct Piece
+        {
+            public Vector2 domain;
+            public Vector2 range;
+            public Func<float, float> transform;
+            public Func<float, float> inverseTransform;
+        }
+
+        // Piecewise function indexed by value ranges.
+        private readonly Dictionary<int, Piece> m_PiecewiseFunctionMap = new Dictionary<int, Piece>();
```

Comment:
The key type `int` is disturbing here. You can directly use `Vector2` with the Dictionary and it will probably void a few bugs.

Dictionary will use the `GetHashCode()` method to bin the values and will use the `IEquatable<Self>.Equals(Self other)` method if `IEquatable<Self>` is implemented, otherwise it will use `object.Equals`.

So it means that if you use directly `Vector2` as Key type, you will achieve the same binning and still have `O(1)` time access, but it will handle equality cases when the `GetHashCode()` values are the same for different `Vector2`  (hash collision).

So other point is about understanding the code, `Vector2` is more explicit then `int` to indicate a range.

Can you please make this change?

Reply:
If I want to push further, I would use a new type pattern enclosing the range:
```
struct ValueRange: IEquatable<ValueRange>
{
    Vector2 m_Range;

    public ValueRange(in Vector2 value) => m_Range = value;
    public Vector2 ToVector2() => m_Range;
    // Insert generated Equality methods by ReSharper here
    
    public static explicit operator Vector2(in ValueRange v) => v.m_Range;
    public static explicit operator ValueRange(in Vector2 v) => new ValueRange(v);
}
```

Pros and cons of the newtype pattern:
Pros:
* The compiler will detect invalid uses of a Vector2 as a ValueRange and inversely. (Type check error)
* It is easier to review
* It helps a lot separating code in small understable and standalone pieces.

Cons:
* In C# it involves a lot of code.

I don't request the use of the newtype pattern as it is quite verbose, it is for your information you may need this pattern in the future.

(But I do request the change to `Vector2` at least :) )

Reply:
Nice, thank you for all this info! Adding it to the toolbelt :) In this case, if it's alright, I have switched to using the `Vector2 `directly as the key. If I have extra time before end of week I may try to add this pattern, but currently need to allocate the remaining time to addressing the various QA and Artist request for the PR)